dynamodb/src/main/scala/monix/connect/dynamodb/DynamoDbSubscriber.scala

```diff
@@ -18,44 +18,67 @@
 package monix.connect.dynamodb
 
 import monix.eval.Task
+import monix.execution.Ack.Continue
+import monix.execution.atomic.AtomicInt
 import monix.execution.cancelables.AssignableCancelable
 import monix.execution.{Ack, Callback, Scheduler}
-import monix.reactive.observers.Subscriber
 import monix.reactive.Consumer
 import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
 import software.amazon.awssdk.services.dynamodb.model.{DynamoDbRequest, DynamoDbResponse}
+import monix.reactive.observers.{SafeSubscriber, Subscriber}
 
+import scala.compat.java8.FutureConverters._
 import scala.concurrent.Future
 
-private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse]()(
+private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse](retriesPerOp: Int)(
   implicit
   dynamoDbOp: DynamoDbOp[In, Out],
   client: DynamoDbAsyncClient)
   extends Consumer[In, Out] {
 
+  require(retriesPerOp >= 1, "Retries per operation must be higher or equal than 1.")
+
   override def createSubscriber(cb: Callback[Throwable, Out], s: Scheduler): (Subscriber[In], AssignableCancelable) = {
-    val sub = new Subscriber[In] {
+    val safeSubscriber = SafeSubscriber {
+      new Subscriber[In] {
 
-      implicit val scheduler = s
-      private var dynamoDbResponse: Task[Out] = _
+        implicit val scheduler = s
+        private var dynamoDbResponse: Task[Out] = _
 
-      def onNext(dynamoDbRequest: In): Future[Ack] = {
-        dynamoDbResponse = Task.from(dynamoDbOp.execute(dynamoDbRequest))
+        def onNext(dynamoDbRequest: In): Future[Ack] = {
+          val retries: AtomicInt = AtomicInt(retriesPerOp)
 
-        dynamoDbResponse.onErrorRecover { case _ => monix.execution.Ack.Stop }
-          .map(_ => monix.execution.Ack.Continue)
-          .runToFuture
-      }
+          dynamoDbResponse = Task.from(dynamoDbOp.execute(dynamoDbRequest))
+          dynamoDbResponse
+            .onErrorHandleWith(ex =>
+              Task.defer {
+                if (retries.get() > 1) {
+                  retries.set(retries.get() - 1)
```

Comment:
These changes provide the user with the possibility of retrying the failed operation.  My question is if `AtomicInt` is being used correctly or would you do that in a different way? (will add a test for that but first thought was good to ask).

On the other hand, was thinking that might also useful to allow to parametrise the parallelisation.

Reply:
It seems a bit weird, it's created per each `onNext` and it is read only once so if will be always true

I don't think we need AtomicInt at all - recursive function should be enough if retries are per operation

Small tip:

```scala
if (retries.get() > 1) {
  retries.set(retries.get() - 1)
  do sth
} else do sth else
```

In concurrent environment, it has a race condition (atomic can be updated between the first get and the set).
You could use `transformAndExtract`, e.g.

```scala
val cond = retries.transformAndExtract { retriesLeft =>
  if (retriesLeft > 1) (true, retriesLeft - 1)
  else (false, retriesLeft)
}
```

or use `getAndDecrement`

Reply:
Thanks you @Avasil, really appreciate your advice! :D 

> It seems a bit weird, it's created per each onNext and it is read only once so if will be always true

What does seems weird? The condition won't be always true, right? It depends on how many times we want to recompute after a failure.

Reply:
> What does seems weird? The condition won't be always true, right? It depends on how many times we want to recompute after a failure.

The method only retries the first failure.
It needs a recursive call instead of:

```scala
Task.deferFuture(toScala(dynamoDbOp.execute(dynamoDbRequest)))
```